docs/notebooks/sdba.ipynb

```diff
@@ -0,0 +1,454 @@
+    "ref = ref.expand_dims(lon=[50, 60, 70, 80, 90, 100])\n",
+    "ref.to_dataset().to_netcdf('reference_data.nc', unlimited_dims=['lon'])\n",
+    "sim.name = 'tas'\n",
+    "sim = sim.expand_dims(lon=[50, 60, 70, 80, 90, 100])\n",
+    "sim.to_dataset().to_netcdf('simulation_data.nc', unlimited_dims=['lon'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "var = \"tas\"\n",
+    "kind = '+'\n",
+    "ref_period = slice('2000', '2015')\n",
+    "sim_period = slice('2000', '2030')\n",
+    "file_ref = 'reference_data.nc'\n",
+    "file_sim = 'simulation_data.nc'\n",
+    "\n",
+    "# We want the robustness of a dayofyear adjustment, but the speed of the monthly computation.\n",
+    "# The compromise is to split the normalization process from the quantile mapping. Smaller groups in normalization\n",
+    "# reduce the boundary effects between months but the monthly quantile mapping is almost as precise in that case.\n",
+    "# This is a scientific decision that should only be taken after careful analysis of the data, it is shown here as\n",
+    "# an example of compromise aimed at accelerating the computation.\n",
+    "g_norm = sdba.Grouper(group='time.dayofyear', window=31)\n",
+    "g_qm = sdba.Grouper(group='time.month')\n",
+    "\n",
+    "# # Step 1 : Normalize hist and ref\n",
+    "maxlon = 6\n",
+    "nlon = 2\n",
+    "\n",
+    "# Normalize (as resample() or groupby()) generates so many small operations that dask struggles\n",
+    "# to even start the computation. Here, with large data, it is more efficient to loop of the\n",
+    "# chunks and to **not** use dask. In the following two loops, \"chunks=\" is not assigned, so\n",
+    "# all data is loaded ad basic numpy operations are used.\n",
+    "for ilon in range(0, maxlon, nlon):\n",
+    "    # Using a `with` statement when opening a file automatically closes it when we exit the context.\n",
+    "    # As multiple open files can be sources of bugs, this helps the coding process\n",
+    "    with xr.open_dataset(file_ref)[var].isel(lon=slice(ilon, ilon + nlon)) as ref:\n",
+    "        da = sdba.processing.normalize(ref, group=g_norm, kind=kind)\n",
+    "        da.name = var\n",
+    "        da.to_netcdf(f\"mydqm_{ilon:03d}_refn.nc\", unlimited_dims=['lon'])\n",
+    "# Hist\n",
+    "for ilon in range(0, maxlon, nlon):\n",
+    "    with xr.open_dataset(file_sim)[var].isel(lon=slice(ilon, ilon + nlon)) as sim:\n",
+    "        da = sdba.processing.normalize(sim.sel(time=ref_period), group=g_norm, kind=kind)\n",
+    "        da.name = var\n",
+    "        da.to_netcdf(f\"mydqm_{ilon:03d}_histn.nc\", unlimited_dims=['lon'])\n",
+    "\n",
+    "# reopen the files\n",
+    "# Here we specify the same chunking for all files, but different values could be used\n",
+    "# depending on the operations or the data.\n",
+    "ref = xr.open_dataset(file_ref, chunks={'lon': 2})[var]\n",
```

Comment:
Starting here, I had multiple instances of `"inconsistent chunks" error` if my chunks weren't perfect multiples of what was saved previously.

For that reason, using `ref = ref.chunk({'lon': 2})` (for example) was far more stable than `xr.open_dataset(file_ref, chunks={'lon': 2})`